File: DesktopSample/Advanced/PageAndRDL/CustomTileProvider/C#/CustomTileProviders/MapTile.cs
using System.IO;
using GrapeCity.ActiveReports.Extensibility.Rendering;
using GrapeCity.ActiveReports.Extensibility.Rendering.Components.Map;

namespace ActiveReports.Samples.CustomTileProviders
{
	/// <summary>
	/// 単一のマップタイルを表します。
	/// </summary>
	internal sealed class MapTile : IMapTile
	{
		/// <summary>
		/// <see cref="MapTile"/>の新しいインスタンスを初期化します。
		/// </summary>
		public MapTile(MapTileKey id, ImageInfo imageStream)
		{
			Id = id;
			Image = imageStream;
		}

		/// <summary>
		/// タイル識別子を取得します。
		/// </summary>
		public MapTileKey Id { get; private set; }

		/// <summary>
		/// タイル画像のストリームを取得します。
		/// </summary>
		public ImageInfo Image { get; private set; }
	}
}

File: DesktopSample/Advanced/PageAndRDL/CustomTileProvider/VB.NET/CustomTileProviders/MapTile.vb
Imports GrapeCity.ActiveReports.Extensibility.Rendering.Components.Map
Imports GrapeCity.ActiveReports.Extensibility.Rendering

	''' <summary>
	''' 単一のマップタイルを表します。
	''' </summary>
	Friend NotInheritable Class MapTile
		Implements IMapTile
		''' <summary>
		''' <see cref="MapTile"/>の新しいインスタンスを初期化します。
		''' </summary>
		Public Sub New(id As MapTileKey, imageStream As ImageInfo)
			_id = id
			_image = imageStream
		End Sub

		''' <summary>
		''' タイル識別子を取得します。
		''' </summary>
		''' 
		Public ReadOnly Property Id() As MapTileKey Implements IMapTile.Id
			Get
				Return _id
			End Get
		End Property
		Private _id As MapTileKey

		''' <summary>
		''' タイル画像のストリームを取得します。
		''' </summary>
		Public ReadOnly Property Image() As ImageInfo Implements IMapTile.Image
			Get
				Return _image
			End Get

		End Property
		Private _image As ImageInfo
	End Class
